internal/pkg/template/template.go

```diff
@@ -63,6 +94,34 @@ func (t *Template) Parse(path string, data interface{}, options ...ParseOption)
 	return &Content{buf}, nil
 }
 
+// UploadEnvironmentCustomResources uploads the environment custom resource scripts.
+func (t *Template) UploadEnvironmentCustomResources(upload func(string, ...CustomResource) (string, error)) ([]CustomResource, error) {
+	return t.uploadCustomResources(upload, envCustomResourceFiles)
+}
+
+func (t *Template) uploadCustomResources(upload func(string, ...CustomResource) (string, error), fileNames []string) ([]CustomResource, error) {
+	customResources := []CustomResource{}
+	for _, name := range fileNames {
+		cr := CustomResource{name: envLambdaZippedScriptName}
+		content, err := t.Read(path.Join(envLambdaRootPath, fmt.Sprintf("%s.js", name)))
+		if err != nil {
+			return nil, err
+		}
+		cr.content = content.Bytes()
+		h := sha256.New()
+		if _, err := h.Write(cr.content); err != nil {
+			return nil, fmt.Errorf("generate SHA for custom resource %s: %w", name, err)
+		}
+		url, err := upload(fmt.Sprintf("%s/%x/%s", scriptDirName, h.Sum(nil), name), cr)
```

Comment:
I wonder if a path like this would make it easier for us to possibly troubleshoot issues?
```
artifacts/custom-resources/dns-cert-validator/<checksum>
```
Like if we want someone to check which version of a lambda they have, this path might be easier to navigate to 🤔 

I don't think it's a one-way door, so not necessary but worth thinking about how we can make debugging easier

Reply:
Good point. I think i'll make it `custom-resources/dns-cert-validator/<checksum>`.